Release briefing 4.6 (#125)
kokkos/kokkos-tutorials: Content/ReleaseBriefings/4_6/Section_BackendUpdates.tex

%==========================================================================

\begin{frame}[fragile]

  {\Huge Backend Updates}

  \vspace{10pt}

\end{frame}

%==========================================================================

% Examples

% note: always keep the [fragile] for your frames!

\begin{frame}[fragile]{CUDA, SYCL and Serial}
  \begin{itemize}

>       \item CUDA: Improved performance for \texttt{Kokkos::parallel\_reduce} on H100 and newer by increasing launch bounds

Reviewer: ```suggestion
      \item CUDA: Improved performance for \texttt{Kokkos::parallel\_reduce} on H100 and newer by removing limitations on the runtime thread configuration
```

Reviewer: Include perf number from Christian's PR

Reviewer: @crtrott said he'd add a plot

>       \item SYCL: Improved sorting performance for non-contiguous views with \texttt{RandomAccessIterator}

Reviewer: ```suggestion
      \item SYCL: Improved sorting performance for non-contiguous views
```
Is that different from the general improvement to the iterators?

Reviewer 2: Yes, we are not falling back to sorting on the host anymore (given a sufficiently new oneDPL version).

>       \item Serial: Reduce fences when using \texttt{Kokkos\_ENABLE\_ATOMICS\_BYPASS}

Reviewer: ```suggestion
      \item Serial: Reduce fences overhead when using \texttt{Kokkos\_ENABLE\_ATOMICS\_BYPASS}
```

  \end{itemize}
\end{frame}

\begin{frame}[fragile]{HIP}
  \begin{itemize}
      \item Change block size deduction to prefer smaller blocks/teams if possible

>       \item Allocate memory with stream ordered semantics (\emph{i.e.}\ use \texttt{hipMallocAsync}) as default

Reviewer: ```suggestion
      \item Allocate memory with stream ordered semantics (\emph{i.e.}\ use \texttt{hipMallocAsync})
```
The option controlling that is IMPL_ only so as far as users are concerned we do `hipMallocAsync`
